packages/rich-text/src/plugins/InsertBeforeFirstVoidBlock/index.tsx

```diff
@@ -0,0 +1,40 @@
+import { KeyboardEvent } from 'react';
+import { PlatePlugin, SPEditor } from '@udecode/plate-core';
+import {
+  isFirstChild,
+  isVoid,
+  getElementFromCurrentSelection,
+  moveToThePreviousLine,
+} from '../../helpers/editor';
+import { Transforms, Path } from 'slate';
+import { BLOCKS } from '@contentful/rich-text-types';
+
+export function createInsertBeforeFirstVoidBlockPlugin(): PlatePlugin {
+  return {
+    onKeyDown: function (editor: SPEditor) {
+      return (event: KeyboardEvent) => {
+        const isEnter = event.key === 'Enter';
+        const [currentElement, currentPath] = getElementFromCurrentSelection(editor);
+        const isFirst = isFirstChild(currentPath as Path);
+
+        if (isEnter && isFirst && isVoid(editor, currentElement)) {
+          const paragraph = {
+            type: BLOCKS.PARAGRAPH,
+            data: {},
+            children: [{ text: '' }],
+          };
+
+          // It adds two paragraphs outside a timeout and after moving the cursor to the previous line, not sure why...
+          setTimeout(() => {
+            Transforms.insertNodes(editor, paragraph, { at: currentPath as Path });
+            moveToThePreviousLine(editor);
+          }, 0);
```

Comment:
Without the timeout:

- Editor -> HR + Paragraph -> Enter **becomes** Editor -> Paragraph + Paragraph (selection) + HR + Paragraph

With the timeout:

- Editor -> HR + Paragraph -> Enter **becomes** Editor -> Paragraph (selection) + HR + Paragraph

Reply:
🤷

Reply:
FYI @cezarsmpio I tried the suggestion from Slack (i.e. `Editor.withoutNormalizing`), it doesn't fix this issue.

Reply:
Ok, thanks!